app/code/Magento/Ui/view/base/web/js/form/form.js

```diff
@@ -339,6 +339,7 @@ define([
          */
         reset: function () {
             this.source.trigger('data.reset');
+            $('[data-index$="_date"] input').val('');
```

Comment:
We must not rely on `input` tag name. Please omit unnecessary quotes `""`.

Reply:
@orlangur updated.